editor/components/icon-button/style.scss

```diff
@@ -14,4 +14,8 @@
 			color: $blue-medium;
 		}
 	}
+
+	span {
+		font-size: 12px;
```

Comment:
Great question to be asking. The icon-button should be just that, an icon button, so any text added should not so much be a label as it should be something to go with the icon itself. In this case, implying a heading size. But it could also be used for quote styles, or possibly even gallery styles (i.e. quote 2 or gallery 3). 

Tying together with your comment https://github.com/WordPress/gutenberg/pull/436#pullrequestreview-32997741, I will open a PR to add a label that makes this clear.